src/plugins/data/server/index_patterns/fetcher/index_patterns_fetcher.ts

```diff
@@ -135,11 +135,20 @@ export class IndexPatternsFetcher {
    */
   async validatePatternListActive(patternList: string[]) {
     const result = await Promise.all(
-      patternList.map((pattern) =>
-        this.elasticsearchClient.count({
-          index: pattern,
-        })
-      )
+      patternList
+        .map((pattern) =>
+          this.elasticsearchClient.count({
+            index: pattern,
+          })
+        )
+        .map((p) =>
+          p.catch((e) => {
```

Comment:
the `count` api will return `count: 0` for a pattern like `nothingbeat-*` but will throw the `index_not_found_exception` on non-wildcard, like `nothingbeat-exact`